libecho/include/EchoProtocol.h

```diff
@@ -42,19 +47,32 @@ class EchoProtocol {
 
     /**
      * @brief Closes a connection.
+     * Connection that was once opened should be closed (even when one of read/write
+     * functions throws an exception).
      */
     void close();
 
     /**
-     * Reads at most count bytes from audio input.
-     * Can receive 0 bytes, but if it doesn't receive any signal for timeout seconds,
-     * it throws an exception.
+     * @brief           Reads at most @p count bytes from audio input and fills buffer with them.
+     * @param buf       pointer to buffer that should be filled
+     * @param count     number of bytes to read
+     * @param timeout   number of seconds; if no signal is detected by then, function assumes
+     *                  that connection is broken;
+     *                  timeout < 0 means that function can wait infinitely
```

Comment:
Warto wyjaśnić co to znaczy "no signal is detected by then". Bo można to interpretować, że przez taki czas druga strona nic nie wyśle (w sensie takim, że użytkownik nie wywoła read), albo że czas przez jaki funkcja próbuje naprawić połączenie przy problemach na niższej warstwie. Jeśli jest to pierwsza interpretacja, to warto opisać w jaki sposób stworzyć długotrwałe połączenie, tak żeby nie zostało zerwane (w sensie że to użytkownik powinien zaimplementować keep-alive)